### PR TITLE
PR for issue #1578 

### LINKED ISSUE #1578: DELETE endpoint returned value not formatted and results in 204
Hi,

Grape `0.19.1`

The return value of a DELETE endpoint is not considered has `@body.present?`. Therefore, the response isn't formatted has expected and it returns a 204. Is it normal ?

The following spec fails :

```ruby
describe 'delete 200, with return value has body' do
      it 'responds to /example' do
        subject.send(:delete, '/example') do
          'deleted'
        end
        send(:delete, '/example')
        expect(last_response.status).to eql 200
        expect(last_response.body).not_to be_empty
      end
    end
```
Thanks

### PR DESCRIPTION
**Pull request by eproulx-petalmd**
Closes #1578 and #1572

### CODE REVIEW

**Comment by dblock** [comment]
I think this is the same as https://github.com/ruby-grape/grape/issues/1572, and we've been wanting a fix, right?

File: lib/grape/endpoint.rb [hunk 1]
```diff
@@ -265,7 +265,10 @@ def run
         run_filters afters, :after
         cookies.write(header)
 
-        # The Body commonly is an Array of Strings, the application instance itself, or a File-like object.
+        #status verifies body presence when DELETE
+        @body ||= response_object if response_object.present?
```
Comment (dblock) [inline lib/grape/endpoint.rb]: I think the `if` is redundant.

**Comment by dblock** [comment]
Run `rubocop -a` then `rubocop --auto-gen-config` to cleanup whatever rubocop is unhappy about.

File: lib/grape/endpoint.rb (hunk 1 above)
Comment (eproulx-petalmd) [inline lib/grape/endpoint.rb]: Indeed, i'll remove it

File: lib/grape/endpoint.rb [hunk 2]
```diff
@@ -265,7 +265,10 @@ def run
         run_filters afters, :after
         cookies.write(header)
 
-        # The Body commonly is an Array of Strings, the application instance itself, or a File-like object.
+        # status verifies body presence when DELETE
+        @body ||= response_object
+
+        # The Body commonly is an Array of Strings, the application instance itself, or a File-like object
         response_object = file || [body || response_object]
```
Comment (dblock) [inline lib/grape/endpoint.rb]: I believe `body` here is the same as `body`, so this should be `file || body`?

File: spec/grape/endpoint_spec.rb [hunk 3]
```diff
@@ -1149,6 +1149,39 @@ def memoized
       end
     end
 
+    describe 'delete 200, with a return value (no explicit body)' do
+      it 'responds to /example delete method' do
+        subject.send(:delete, '/example', anchor: false) do
```
Comment (dblock) [inline spec/grape/endpoint_spec.rb]: I don't think you need to `send` these, `delete` is public, so `subject.delete ...`

File: spec/grape/endpoint_spec.rb [hunk 4]
```diff
@@ -1149,6 +1149,39 @@ def memoized
...
+          'deleted'
+        end
+        send(:delete, '/example/')
```
Comment (dblock) [inline spec/grape/endpoint_spec.rb]: Same, just `delete '/example'`, these are rack helpers.

File: lib/grape/endpoint.rb (hunk 2 above)
Comment (eproulx-petalmd) [inline lib/grape/endpoint.rb]: Yes, it's the same. I forgot to remove it. On it 👍 
Thanks

**Comment by dblock** [comment]
You need a CHANGELOG and it's good to go. Thx.

**Comment by dblock** [comment]
Merged, thanks.

**Pull request merged**